### DISCUSSION TITLE
Shelly + wall switch

### DISCUSSION THREAD

**Discussion by mafyata** [RavenSystem/esp-homekit-devices#1783] [MEPLHAA Scripts]
Hi, I will set up a new home. I want to control my wall switches(2 gang) with my iPhone. I want to buy Shelly relays(1, 1L, 1pm, 2pm…) and flash them with HAA.

1. Can you tell me which relays to buy and the json config for them?
2. If I turn off the lights from the wall switch, will I be able to turn on the lights from my phone or the wall switch must be always at ON position?
![IMG_00641](https://user-images.githubusercontent.com/48009267/157189048-e71a29bb-943f-4387-8eb9-35b1d562b2af.jpg)

**Comment by xrust83** [comment]
you can use parallel switch and homekit.  To do this, you need to use a toggle switch.  I did it on Sonoff Dual, but the principle is the same for everyone, but with the haa installation, the difference is mainly in the shape and size of the box

https://youtu.be/BmMQ_jL7oDY
https://youtu.be/MANoTO2CaOA
https://github.com/RavenSystem/esp-homekit-devices/wiki/Devices-Database

**Reply by mafyata** [reply]
If I use Shelly Plus 2PM, can I connect the wall swich to S1 and S2? Will it be perfect solution or the wall swich must be always at ON position? My idea is when I turn OFF the lights from the wall switch, after that to be able to turn ON lights from the phone.

**Reply by xrust83** [reply]
Yes, it will be visibility.  s1 and s2 are set to change state.  The switch will work on the changeover box.  can be both up and down and on can be both up and down.  depends on what state the light was in before re-clipping.  the switch becomes just a trigger.  but for this you need to correctly assemble the circuit, I hope you understand this, this is important.  I think sonoff mini is a copy of your shelly.  it’s tedious to clarify, respectively, GPIO, they may differ from the manufacturer

**Reply by mafyata** [reply]
Is this correct?

![aaa](https://user-images.githubusercontent.com/48009267/157251080-f5ea8299-2ecd-45be-b57a-a9de457f3586.jpg)

**Reply by xrust83** [reply]
shelly is different from sonoff mini.  on the mini contacts of the low-current buttons, on the shell they are connected to the phase.  sonoff are safe in this regard, since there is no high-voltage voltage on the switch.  your schema is correct

**Reply by andyallcorn** [reply]
> 

Just to be clear, the Shelly **Plus** models aren’t compatible with HAA. The original models (such as 1PM, 2.5, etc) are. I’m using a number of these in wall switches, to allow dual control.

**Reply by mafyata** [reply]
Thank you. I was planning to buy “plus” models.
